drivers/lora/sx1276.c

```diff
@@ -0,0 +1,717 @@
+{
+	int ret, i;
+	u8_t regval;
+
+	for (i = 0; i < data_len; i++) {
+		ret = sx1276_read(dev, SX1276_REG_FIFO, &regval, 1);
+		if (ret < 0) {
+			LOG_ERR("Unable to read FIFO");
+			return -EIO;
+		}
+
+		data[i] = regval;
+	}
+
+	return 0;
+}
+
+static int sx1276_lora_recv(struct device *dev, u8_t *data)
+{
+	struct sx1276_data *dev_data = dev->driver_data;
+	int ret;
+	u8_t regval, len, loc = 0;
+
+	/* Clear all flags */
+	ret = sx1276_write(dev, SX1276_REG_IRQ_FLAGS, 0xFF);
+	if (ret < 0) {
+		LOG_ERR("Unable to write IRQ_FLAGS");
+		return -EIO;
+	}
+
+	/* Set DIO0 to Rx done */
+	ret = sx1276_read(dev, SX1276_REG_DIO_MAPPING1, &regval, 1);
+	if (ret < 0) {
+		LOG_ERR("Unable to read DIO_MAPPING1");
+		return -EIO;
+	}
+
+	regval &= ~REG_DIO_MAPPING1_DIO0_MASK;
+	ret = sx1276_write(dev, SX1276_REG_DIO_MAPPING1, regval);
+	if (ret < 0) {
+		LOG_ERR("Unable to write DIO_MAPPING1");
+		return -EIO;
+	}
+
+	ret = sx1276_read(dev, SX1276_REG_OPMODE, &regval, 1);
+	if (ret < 0) {
+		LOG_ERR("Unable to read OPMODE");
+		return -EIO;
+	}
+
+	/* FIFO cannot be accessed in sleep mode */
+	if (regval & SX1276_OPMODE_MODE_SLEEP) {
+		regval &= ~SX1276_OPMODE_MODE_MASK;
+		regval |= SX1276_OPMODE_MODE_STDBY;
+	}
+
+	ret = sx1276_write(dev, SX1276_REG_OPMODE, regval);
+	if (ret < 0) {
+		LOG_ERR("Unable to write OPMODE");
+		return -EIO;
+	}
+
+	/* Make use of full FIFO for Rx */
+	ret = sx1276_write(dev, SX1276_REG_FIFO_RX_BASE_ADDR, 0x00);
+	if (ret < 0) {
+		LOG_ERR("Unable to write FIFO Rx base addr");
+		return -EIO;
+	}
+
+	ret = sx1276_write(dev, SX1276_REG_FIFO_ADDR_PTR, 0x00);
+	if (ret < 0) {
+		LOG_ERR("Unable to write FIFO addr pointer");
+		return -EIO;
+	}
+
+	ret = sx1276_read(dev, SX1276_REG_OPMODE, &regval, 1);
+	if (ret < 0) {
+		LOG_ERR("Unable to read OPMODE");
+		return -EIO;
+	}
+
+	regval &= ~SX1276_OPMODE_MODE_MASK;
+	regval |= SX1276_OPMODE_MODE_RXCONTINUOUS;
+	ret = sx1276_write(dev, SX1276_REG_OPMODE, regval);
+	if (ret < 0) {
+		LOG_ERR("Unable to write OPMODE");
+		return -EIO;
+	}
+
+	k_sem_take(&dev_data->data_sem, K_FOREVER);
+
+	ret = sx1276_read(dev, SX1276_REG_RX_NB_BYTES, &len, 1);
+	if (ret < 0) {
+		LOG_ERR("Unable to read RX len");
+		return -EIO;
+	}
+
+	ret = sx1276_read(dev, SX1276_REG_FIFO_RX_CUR_ADDR, &loc, 1);
+	if (ret < 0) {
+		LOG_ERR("Unable to read RX curr addr");
+		return -EIO;
+	}
+
+	ret = sx1276_write(dev, SX1276_REG_FIFO_ADDR_PTR, loc);
+	if (ret < 0) {
+		LOG_ERR("Unable to write FIFO Tx addr pointer");
+		return -EIO;
+	}
+
+	ret = sx1276_fifo_read(dev, data, len);
+	if (ret < 0) {
+		LOG_ERR("Unable to read RX data");
+		return -EIO;
+	}
+
+	return len;
+}
+
+static int sx1276_lora_send(struct device *dev, u8_t *data, u32_t data_len)
+{
+	int ret;
+	u8_t regval;
+
+	/* Set DIO0 to Tx done */
+	ret = sx1276_read(dev, SX1276_REG_DIO_MAPPING1, &regval, 1);
+	if (ret < 0) {
+		LOG_ERR("Unable to read DIO_MAPPING1");
+		return -EIO;
+	}
+
+	regval &= ~REG_DIO_MAPPING1_DIO0_MASK;
+	regval |= 0x1 << 6;
```

Comment:
Can you change this from magic numbers into defines like the other register values?